maat-court-data-api/src/test/java/gov/uk/courtdata/applicant/controller/ApplicantControllerTest.java

```diff
@@ -56,16 +57,63 @@ void givenAValidationException_whenGetRepOrderApplicantLinksIsInvoked_thenCorrec
         verify(repOrderApplicantLinksService).find(REP_ID);
     }
 
+    @Test
+    void givenValidRequest_whenUpdateRepOrderApplicantLinkIsInvoked_thenUpdateIsSuccess() throws Exception {
+        RepOrderApplicantLinksDTO repOrderApplicantLinksDTO = TestModelDataBuilder.getRepOrderApplicantLinksDTO(ID);
+        when(repOrderApplicantLinksService.update(any())).thenReturn(repOrderApplicantLinksDTO);
+        mvc.perform(MockMvcRequestBuilders.put(ENDPOINT_URL + "/rep-order-applicant-links")
+                        .content(objectMapper.writeValueAsString(repOrderApplicantLinksDTO))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void givenAEmptyContent_whenUpdateRepOrderApplicantLinkIsInvoked_thenCorrectErrorResponseIsReturned() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.put(ENDPOINT_URL + "/rep-order-applicant-links")
+                        .content("{}")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void givenInValidRequest_whenUpdateRepOrderApplicantLinkIsInvoked_thenCorrectErrorResponseIsReturned() throws Exception {
+        when(repOrderApplicantLinksService.update(any())).thenThrow(new RequestedObjectNotFoundException("Rep Order Applicant Links not found"));
+        mvc.perform(MockMvcRequestBuilders.put(ENDPOINT_URL + "/rep-order-applicant-links")
+                        .content(objectMapper.writeValueAsString(TestModelDataBuilder.getRepOrderApplicantLinksDTO(ID)))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is(404));
+    }
+
+    @Test
+    void givenInternalServerError_whenUpdateRepOrderApplicantLinkIsInvoked_thenCorrectErrorResponseIsReturned() throws Exception {
+        when(repOrderApplicantLinksService.update(any())).thenThrow(EmptyResultDataAccessException.class);
+        mvc.perform(MockMvcRequestBuilders.put(ENDPOINT_URL + "/rep-order-applicant-links")
+                        .content(objectMapper.writeValueAsString(TestModelDataBuilder.getRepOrderApplicantLinksDTO(ID)))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is5xxServerError())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.code").value(ErrorCodes.DB_ERROR));
+    }
+
+    @Test
+    void givenCorrectId_whenGetApplicantHistoryIsInvoked_thenResponseIsReturned() throws Exception {
+        ApplicantHistoryDTO applicantHistoryDTO = TestModelDataBuilder.getApplicantHistoryDTO(ID, "N");
+        when(applicantHistoryService.find(ID)).thenReturn(applicantHistoryDTO);
+        mvc.perform(MockMvcRequestBuilders.get(ENDPOINT_URL + "/applicant-history/" + ID))
+                .andExpect(status().isOk());
+        verify(applicantHistoryService).find(1);
```

Comment:
If we've mocked the call with `when`, then there's no need for a `verify` call here.